### PR TITLE
SMD-710 - Pathfinders - validate postcodes

### PR DESCRIPTION
**Pull request by wjrm500**
### Ticket

[SMD-710 - Pathfinders - validate postcodes](https://dluhcdigital.atlassian.net/browse/SMD-710)

### Change summary
Add validation to check for postcode or comma-separated list of postcodes in postcodes field in "Project location" table from new Pathfinder Reporting Template.

### CODE REVIEW

**Review by Joel-Pearce — CHANGES_REQUESTED** [review]

File: tables/checks.py [hunk 1]
```diff
@@ -67,3 +68,21 @@ def max_word_count(element, *, max_words):
...
+@pa.extensions.register_check_method(check_type="element_wise")
+def postcode_list(element):
+    """Checks that a string can be split on commas and each element matches a basic UK postcode regex.
+
+    :param element: an element to check
+    :return: True if passes the check, else False
+    """
+    if not isinstance(element, str):
+        raise TypeError("Value must be a string")
+    postcode_regex = r"^[A-Z]{1,2}\d[A-Z\d]? ?\d[A-Z]{2}$"
```
Comment (nsnape) [inline tables/checks.py:82]: This is slightly different to the regex here https://github.com/communitiesuk/funding-service-design-post-award-data-store/blob/71b823014f2f02340da8c3271ee4cff69721c8a8/core/transformation/utils.py#L9

Should they be the same? Or use the same source?

**Review by nsnape — CHANGES_REQUESTED** [review]

File: tables/checks.py (hunk 1 above)
Comment (wjrm500) [reply tables/checks.py:82]: Existing postcode regex allows this postcode, which new regex marked as invalid: `SW1A 1AA1`. Based on the fact that this postcode is invalid, I will replace the existing POSTCODE_REGEX const with the new one and use this in `checks.py`.

image

Comment (wjrm500) [reply tables/checks.py:82]: Scratch that last message, changing `POSTCODE_REGEX` causes a bunch of tests to fail. I think best to revert to using existing regex regardless of whether or not it is objectively more accurate than the new one!

**Review by Joel-Pearce — APPROVED** [review]
LGTM

**Review by nsnape — APPROVED** [review]

**Pull request merged**